### DISCUSSION TITLE
Requesting getting information about the counterpart

### DISCUSSION THREAD

**Discussion by polvallverdu** [sigma67/ytmusicapi#443] [Q&A] [answered]
I'm writing a script to get the counterpart of a song, since I would like this feature here, but I don't really know what the best way to implement it to the lib would be, so I'll post all the info that I've got. 

The call should be to `youtubei/v1/next` with the header `videoId`, and this would return info about the counterpart video Id, and segments to match the audio within the transition. Info should be here

```ts
const counterpart =
	data.contents.singleColumnMusicWatchNextResultsRenderer.tabbedRenderer
		.watchNextTabbedResultsRenderer.tabs[0].tabRenderer.content.musicQueueRenderer
		.content.playlistPanelRenderer.contents[0].playlistPanelVideoWrapperRenderer
		.counterpart[0]
const counterpartVideoId =
	counterpart.counterpartRenderer.playlistPanelVideoRenderer.videoId
```

This would only work if the account logged in has a youtube premium membership.

**Comment by sigma67** [comment, marked as answer]
https://ytmusicapi.readthedocs.io/en/stable/search.html?q=counterpart&check_keywords=yes&area=default
